core/shared/src/main/scala/zio/prelude/AssociativeCompose.scala

```diff
@@ -45,21 +46,24 @@ object AssociativeCompose {
   }
 
   implicit val URIOIdentityCompose: IdentityCompose[URIO] = new IdentityCompose[URIO] {
-    def identity[A]: URIO[A, A] = URIO.environment
+    def identity[A]: URIO[A, A] = URIO.service[A]
```

Comment:
For example, this gives
```
zio-prelude/core/shared/src/main/scala/zio/prelude/AssociativeCompose.scala:49:47: could not find implicit value for izumi.reflect.Tag[A]. Did you forget to put on a Tag, TagK or TagKK context bound on one of the parameters in A? e.g. def x[T: Tag, F[_]: TagK] = ...
[error] 
[error] 
[error] <trace>: 
[error]   deriving Tag for A, dealiased: A:
[error]   could not find implicit value for Tag[A]: A is a type parameter without an implicit Tag!
[error]     def identity[A]: URIO[A, A] = URIO.service[A]
[error]                                               ^
```

Reply:
Yeah `URIO[A, A]` is not an identity here since it accesses a specific service from the environment. It is basically `ZEnvironment[A] => A` at this point and is only defined if there is a tag for `A` and `A` is a single service. `URIO[A, ZEnvironment[A]]` would be closer to an identity since that would be conceptually `ZEnvironment[A] => ZEnvironment[A]`.